### PR TITLE
Fix Markdown link checking action

### LINKED ISSUE #170: Markdown link checking isn't working
The current config breaks the checking, and makes the tool report status 0 for all links (w/o checking anything). For proof see:

- #169

/cc @svrnm

### PR DESCRIPTION
**Pull request by tpaschalis**
Fixes #170

### CODE REVIEW

File: specification.md [hunk 1]
```diff
@@ -3144,14 +3144,14 @@ reduce the number of connections to the Server when a very large number
 
 ### Configuration Management
 
-* [Uber Flipr](https://eng.uber.com/flipr/)
+* [Uber Flipr](https://web.archive.org/web/20240223111552/https://www.uber.com/blog/flipr/)
```
Comment (tpaschalis) [inline specification.md:3147]: The bare link is now https://www.uber.com/blog/flipr/, but hitting the link with cURL results in `406 Not Acceptable` so I opted for the web.archive.org link.

File: specification.md [hunk 2]
```diff
@@ -3144,14 +3144,14 @@ reduce the number of connections to the Server when a very large number
...
 * Facebook's
   [Holistic Configuration Management](https://research.facebook.com/file/877841159827226/holistic-configuration-management-at-facebook.pdf)
   (push)
 
 ### Security and Certificate Management
 
-* [mTLS in Go](https://kofo.dev/how-to-mtls-in-golang)
+* [mTLS in Go](https://web.archive.org/web/20240223142649/https://kofo.dev/how-to-mtls-in-golang)
```
Comment (tpaschalis) [inline specification.md:3154]: This person's website has gone offline so I replaced this with the link from the Wayback Machine.

File: specification.md (hunk 1 above)
Comment (chalin) [reply specification.md:3147]: Given that https://www.uber.com/blog/flipr exists, I'd vote to have an ignore pattern for this one.

**Review by chalin — COMMENTED** [review]
Thanks for figuring this out! See inline comment.

File: specification.md (hunk 1 above)
Comment (tpaschalis) [reply specification.md]: Ok, done in https://github.com/open-telemetry/opamp-spec/pull/196/commits/515ba8a89e7596121e2de554e9532550f9baac30

**Review by tigrannajaryan — APPROVED** [review]

**Comment by tigrannajaryan** [comment]
Thank you @tpaschalis

**Pull request merged**